src/util/AsFunctions.ts

```diff
@@ -12,19 +13,20 @@ function asString(input: any): string {
 	let retval: string = null;
 	if(isDefined(input)) {
 		const tOf: string = typeof input;
-		switch(tOf) {
-			case "string":
+		switch(typeof input) {
+			case JSType.STR:
 				retval = input;
 				break;
-			case "function":
-			case "object":
-				retval = JSON.stringify(input);
-				break;
-			case "symbol":
-				retval = input.toString();
+			case JSType.BOOL:
+			case JSType.BIGINT:
+			case JSType.NUM:
+				retval = input + "";
 				break;
+			case JSType.SYM:
+			case JSType.FN:
+			case JSType.OBJ:
 			default:
-				retval = input + "";
+				retval = JSON.stringify(input);
```

Comment:
This breaks the use of the core JS concept of the toString() method on Object.prototype that allows for polymorphic generation of strings.  JSON conversion might be useful in some cases, but it would need to be predicated on a much more narrow situation being present.

Reply:
Two separate methods now present: asString and asJSON